CONTRIBUTING.md

```diff
@@ -1,47 +1,104 @@
+
+This document assumes that you already know:
+
+- How to [create a Nix package](https://nix.dev/tutorials/packaging-existing-software#).
+- How to [use GitHub and Git](https://docs.github.com/en/get-started/quickstart/hello-world).
+
+In addition, a GitHub account is required, which you can sign up for [here](https://github.com/signup).
+
+## How to create pull requests
+
+Packagers are encouraged to contribute NGI projects to Nixpkgs, instead of to this repository. However, there are many reasons for not being able to do so, including:
+
+- Expediting the public availability of a package prior to its acceptance into Nixpkgs and landing in a channel.
+- Package is not a [good candidate for Nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package).
+- Package is a good candidate for Nixpkgs, but no one is willing to be a maintainer :cry:.
+
+In any case, it is encouraged to create a pull request to Nixpkgs, then to this repository, with a comment linking to the pull request to Nixpkgs in the description and the Nix expressions.
+
+---
+
+Now that this is out of the way. To create pull requests to NGIpkgs:
+
+1. Set up a local version of NGIpkgs. If you don't have write access to NGIpkgs, create a [fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo#forking-a-repository) of the repository and clone it.
+
+   ```ShellSession
+   git clone https://github.com/some-owner/ngipkgs.git
+   cd ngipkgs
+   ```
+
+1. Create and switch to a new Git branch.
+
+   ```ShellSession
+   git checkout -b some-branch
+   ```
+
+1. To add a package, start by creating a `package.nix` file in the package directory `pkgs/by-name/some-package`, where `some-package` will be the package attribute name.
+
+   ```ShellSession
+   mkdir -p pkgs/by-name/some-package
+   some-editor pkgs/by-name/some-package/package.nix
+   ```
+
+   Make sure to:
+
+   - Test the package on `x86_64-linux`.
+
+     ```ShellSession
+     git add pkgs/by-name/some-package
+     nix build .#some-package
+     ```
+
+   - Format the Nix expressions with [Alejandra](https://github.com/kamadorueda/alejandra).
+
+     ```ShellSession
+     alejandra pkgs/by-name/some-package
+     ```
+
+   To see how it’s done, an existing example is [libgnunetchat](https://github.com/ngi-nix/ngipkgs/blob/main/pkgs/by-name/libgnunetchat/package.nix).
+
+1. To add a module, start by creating a `default.nix` file in the module directory `projects/some-module`, where `some-module/service` will be the module name.
+
+   ```shellSession
+   mkdir -p projects/some-module
+   some-editor projects/some-module/default.nix
+   ```
+
+   Make sure to:
+
+   - Add the module options in `service.nix`.
+
+   - Add the module tests in `test.nix`, or under a test directory, and expose them through `some-package.passthru.tests.some-test`, where `some-package` is a package attribute name.
```

Comment:
Here we should mention that the way to access the NixOS test derivation in the package derivation definition is via `nixosTests.${projectName}`, see e.g.(

https://github.com/ngi-nix/ngipkgs/blob/a88709835a182babe2c7fbcf4db5424ebb735351/pkgs/by-name/mcaptcha/package.nix#L200